src/storage-client/src/controller.rs

```diff
@@ -1299,30 +1357,58 @@ where
             if desc.sink.from != *from_id {
                 return Err(StorageError::SourceIdReused(*id));
```

Comment:
I know this isn't you, but this seems more like an input inconsistency than ID reuse... am I understanding that right?